### ISSUE TITLE
Enable user to detect launch options run time

### ISSUE THREAD

**Issue by abhinaba-ghosh** [microsoft/playwright#2570]
It would be nice to have an option to detect launch option run time. 

Suppose, I am launching Playwright like:

```js
 const browser = await playwright['chromium'].launch({
      args: ['--remote-debugging-port=9222'],
    });
```
Now, I want to fetch the launch options anywhere from my code like:

```js
await browser.getLaunchOptions() // returns {args:[]}
```

**Comment by pavelfeldman** [comment]
Playwright is a library and it assumes you are in the control of both launching and querying. If that is the case, you could store this information yourself:

```js
const argsSymbol = Symbol('args');
...
...
const args = ['--remote-debugging-port=9222'];
const browser = await playwright['chromium'].launch({ args });
(browser as any)[argsSymbol] = args;
```

**Comment by abhinaba-ghosh** [comment]
Hi @pavelfeldman, I agree. 

But, I was doing a comparative study between playwright and cypress for some ease of handling browser instances. I found that cypress is having a [similar capability](https://docs.cypress.io/api/plugins/browser-launch-api.html#Modify-browser-launch-arguments-preferences-and-extensions) out of the box. So, I was curious to know if, in Playwright, we can have the same or not. 

Basically, I have published [playwright-lighthouse](https://www.npmjs.com/package/playwright-lighthouse) package, where user needs to send `remote-debugging-port` as an input. I was thinking if there is a way to fetch that port information in time. Then there will be some relief from the user end. 

Thoughts?

**Comment by pavelfeldman** [comment]
Check out our `launchServer` API. You can launch a single server, then connect to it from both Playwright and pass its wsEndpoint port to Lighthouse.

**Comment by pavelfeldman** [comment]
I hope this worked for you! Closing!

**Comment by privatenumber** [comment]
@pavelfeldman 

I tried the following as per your suggestion:

```js
(async () => {
	const browserServer = await chromium.launchServer();
	const wsEndpoint = browserServer.wsEndpoint();
	const [, port] = wsEndpoint.match(/:(\d+)\//);

	const runnerResult = await lighthouse('https://google.com', { port });

	console.log(runnerResult);
})();
```

I get the following error:
```
Error: Protocol JSON API error (list), status: 426
    at IncomingMessage.<anonymous> (/project/node_modules/.pnpm/lighthouse@7.0.1/node_modules/lighthouse/lighthouse-core/gather/connections/cri.js:107:18)
    at IncomingMessage.emit (events.js:327:22)
    at endReadableNT (internal/streams/readable.js:1327:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
```

Any idea how to work around this?

**Comment by cian-sheehy** [comment]
@pavelfeldman

I am also getting the same error when trying to connect to lighthouse. Any ideas why this is happening?
